### DISCUSSION TITLE
User-readable debug data

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#26412] [Suggest an Idea]
### Tell us more.

Ref: https://github.com/renovatebot/renovate/discussions/25913#discussioncomment-7899576

We currently record some data such as the version of Renovate the PR was created or updated in which could be more useful for users if either:
- Hidden, but plain text, or
- Displayed in markdown (e.g. behind a collapsed details dropdown)

In future this might also include the job (log) context.

There's some other data we need and will add in future which is probably less useful to users, such as:
- The target branch at time of PR creation, or
- Labels or label hash at time of PR creation

So maybe we should have hidden and non-hidden? Also we should consider what to do for platforms without html details dropdowns - should we display them by default, or hide them?

**Comment by secustor** [comment]
Labels at creation time would allow us to detect changed labels. See #9950

What I would like to see too is adding the `traceId` and `spanId`. That way admins can pinpoint the run in their Observability tool.

**Reply by rarkins** [reply]
I don't think that "labels used at PR creation time" needs to be user visible though - those can remain as hidden debug data?

**Reply by rarkins** [reply]
@RahulGautamSingh was working on labels

**Reply by secustor** [reply]
> I don't think that "labels used at PR creation time" needs to be user visible though - those can remain as hidden debug data?

I have meant to add this as hidden data.

**Reply by RahulGautamSingh** [reply]
Check this out: https://github.com/renovatebot/renovate/pull/25340
